packages/trader/src/Stores/Modules/Trading/Actions/__tests__/duration.spec.ts

```diff
@@ -0,0 +1,282 @@
+                                    contract_type: 'ACCU',
+                                    expiry_type: 'no_expiry',
+                                    growth_rate_range: [0.01, 0.02, 0.03, 0.04, 0.05],
+                                    high_barrier: barrier_intraday,
+                                    low_barrier: barrier_intraday,
+                                    max_contract_duration: '0',
+                                    min_contract_duration: '0',
+                                    sentiment: 'low_vol',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_daily,
+                                    barrier_choices: barrier_choices_daily,
+                                    contract_display: 'Vanilla Long Call',
+                                    contract_type: 'VANILLALONGCALL',
+                                    expiry_type: 'daily',
+                                    max_contract_duration: '365d',
+                                    min_contract_duration: '1d',
+                                    sentiment: 'up',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_daily,
+                                    barrier_choices: barrier_choices_daily,
+                                    contract_display: 'Vanilla Long Put',
+                                    contract_type: 'VANILLALONGPUT',
+                                    expiry_type: 'daily',
+                                    max_contract_duration: '365d',
+                                    min_contract_duration: '1d',
+                                    sentiment: 'down',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_intraday,
+                                    barrier_choices: barrier_choices_intraday,
+                                    contract_display: 'Vanilla Long Call',
+                                    contract_type: 'VANILLALONGCALL',
+                                    expiry_type: 'intraday',
+                                    max_contract_duration: '1d',
+                                    min_contract_duration: '1m',
+                                    sentiment: 'up',
+                                },
+                                {
+                                    ...vanilla_contracts_for_data,
+                                    barrier: barrier_intraday,
+                                    barrier_choices: barrier_choices_intraday,
+                                    contract_display: 'Vanilla Long Put',
+                                    contract_type: 'VANILLALONGPUT',
```

Comment:
I guess, if you are using TRADE_TYPES.VANILLA.CALL below, it can be also applied here and on lines 90,79, 68, 54 
Please, ignore if not relevant

Reply:
@kate-deriv unfortunately, using external variables is prohibited inside jest.mocks

Reply:
Oh, sorry I didn't know